editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-property-strategies.ts

```diff
@@ -259,6 +260,92 @@ export const setZIndexOnPastedElement =
     ])
   }
 
+export const convertFragmentLikeChildrenToVisualSize =
+  (
+    elementToReparent: ElementPathSnapshots,
+    metadata: MetadataSnapshots,
+    oldAllElementProps: AllElementProps,
+    childPathLookup: OldPathToNewPathMapping,
+    newParent: ElementPath,
+    reparentStrategy: ReparentStrategy,
+    projectContents: ProjectContentTreeRoot,
+    openFile: string | null | undefined,
+    propertyStrategies: Array<
+      (
+        elementToReparent: ElementPathSnapshots,
+        metadata: MetadataSnapshots,
+        newParent: ElementPath,
+      ) => ReparentPropertyStrategy
+    >,
+  ): ReparentPropertyStrategy =>
+  () => {
+    const isElementFragmentLike = treatElementAsFragmentLike(
+      metadata.originalTargetMetadata,
+      oldAllElementProps,
+      metadata.originalPathTrees,
+      elementToReparent.oldPath,
+    )
+    if (!isElementFragmentLike) {
+      return left('Element is not fragment-like')
+    }
+
+    const childPaths = replaceFragmentLikePathsWithTheirChildrenRecursive(
+      metadata.originalTargetMetadata,
+      oldAllElementProps,
+      metadata.originalPathTrees,
+      [elementToReparent.oldPath],
+    )
+
+    const commands = childPaths.flatMap((originalPath) => {
+      const instance = MetadataUtils.findElementByElementPath(
+        metadata.originalTargetMetadata,
+        originalPath,
+      )
+      const newPath = childPathLookup[EP.toString(originalPath)]
+      if (instance == null || newPath == null) {
+        return []
+      }
+      let baseLayoutConversionCommands: Array<CanvasCommand> = []
+      if (reparentStrategy === 'REPARENT_AS_ABSOLUTE') {
+        baseLayoutConversionCommands = getAbsoluteReparentPropertyChanges(
```

Comment:
Thinking out loud and I'm genuinely not sure: I wonder if this should come from the strategy.

Reply:
"this" being `reparentStrategy`?

Reply:
Ah no, the invocation of `getAbsoluteReparentPropertyChanges`, just because it looks like strategy specific behaviour. But no need to change anything for this PR!